src/type/validate.js

```diff
@@ -185,8 +204,129 @@ function validateDirectives(context: SchemaValidationContext): void {
           [getDeprecatedDirectiveNode(arg.astNode), arg.astNode?.type],
         );
       }
+
+      if (arg.defaultValue) {
+        validateDefaultValue(
+          context,
+          `@${directive.name}(${arg.name}:)`,
+          arg.type,
+          arg.defaultValue,
+          arg.astNode?.defaultValue,
+        );
+      }
+    }
+  }
+}
+
+function validateDefaultValue(
+  context: SchemaValidationContext,
+  coordinate: string,
+  type: GraphQLInputType,
+  defaultValue: GraphQLDefaultValueUsage,
+  defaultValueNode: ?ConstValueNode,
+): void {
+  if (defaultValue.literal) {
+    validateInputLiteral(
+      defaultValue.literal,
+      type,
+      undefined,
+      (error, path) => {
+        context.reportError(
+          `${coordinate} has invalid default value${printPathArray(path)}: ${
+            error.message
+          }`,
+          error.nodes,
+        );
+      },
+    );
+  } else {
+    let errors = [];
+    validateInputValue(defaultValue.value, type, (error, path) => {
+      errors.push([error, path]);
+    });
+
+    // If there were validation errors, check to see if it can be "uncoerced"
+    // and then correctly validated. If so, report a clear error with a path
+    // to resolution.
+    if (errors.length > 0) {
+      try {
+        const uncoercedValue = uncoerceDefaultValue(defaultValue.value, type);
+        const uncoercedErrors = [];
+        validateInputValue(uncoercedValue, type, (error, path) => {
+          uncoercedErrors.push([error, path]);
+        });
+        if (uncoercedErrors.length === 0) {
+          context.reportError(
+            `${coordinate} has invalid default value: ${inspect(
+              defaultValue.value,
+            )}. Did you mean: ${inspect(uncoercedValue)}?`,
```

Comment:
Is it worth considering increasing the `{depth:...}` value for these `inspect` calls by one or two levels?

Reply:
This is our version of `inspect`, not Node's, and it always prints complete values, no `depth` option.